### PR TITLE
Integrate Foundation for Apps framework

### PR DESCRIPTION
**Pull request by myabc**
https://community.openproject.org/work_packages/17751

Begins a major CSS refactor.
- [x] Layout fixes (major issues)
- [x] News fixes
- [x] Buttons
- [x] Update OP themes
- [x] Update OP plugins
  - [x] Backlogs
  - [x] Costs
  - [x] Reporting (Engine)
  - [x] My Project Page
  - [x] Mobile OTP
  - [x] Meetings
- [x] Typography (normalisation)
- [x] Basic Form fixes
  - [x] General
- [x] Pagination fixes

Moved to other PRs:
- Filter design improvements
- Button groups, segmented buttons (in toolbar)
- Progress bars
- Form rework:
  - File upload
  - Modals

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://feature-foundation-apps-framework.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

**Comment by myabc** [comment]
@ulferts @0xF013 this is ready for review. I've had to shrink the scope and break some subtasks out into other PRs/WPs.

File: app/assets/stylesheets/default.css.sass [hunk 1]
```diff
@@ -33,12 +33,17 @@
 //= require_self
 //= require legacy/main
```
Comment (ulferts) [inline app/assets/stylesheets/default.css.sass]: How does this relate to https://github.com/opf/openproject/pull/2353? Will the menu on wiki et. al. when using themes be lost again.

Comment (myabc) [inline app/assets/stylesheets/default.css.sass]: @ulferts the changes from #2353 didn't make it into dev.

**Comment by ulferts** [comment]
The forms in various places are broken:

WP Filter:
image

Projects administration:
image

Users administration:
image

Timelines
image

Timelines administration:
image

image

Calendar:
image

**Comment by myabc** [comment]
> in various places are broken:

see my comment on the WP. I don't want to address form sizing (which would mean adding a bunch of ad-hoc hacks/overrides) until I address text sizing overall.

**Comment by ulferts** [comment]
Action menus have discs before each item:

image

File: app/assets/stylesheets/default.css.sass (hunk 1 above)
Comment (ulferts) [inline app/assets/stylesheets/default.css.sass]: I understand that. That will then be addressed in a separate PR I guess?

**Comment by myabc** [comment]
@ulferts 

> Action menus have discs before each item:

should be fixed in 207a396.

File: app/assets/stylesheets/default.css.sass (hunk 1 above)
Comment (myabc) [inline app/assets/stylesheets/default.css.sass]: @ulferts yes. Better theming is beyond the scope of this PR.

**Comment by myabc** [comment]
@ulferts will address Rubocop warnings in TU tests in a separate PR.

**Comment by myabc** [comment]
@ulferts I should have now addressed all of the issues you highlighted, with the exception of _Timelines administration page_.
- Work package filters have received a big overhaul.
- Other pages/views have more minimal changes:
  - Projects administration
  - Users administration
  - Timelines
  - Calendar

Hopefully, there'll be time to improve these views in a future PR.

**Comment by ulferts** [comment]
I like the new filter layout for work packages!

I have looked through the application and made screenshots of everything that looked weird to me. Lets talk about what needs to be fixed before merging:

Positioning of the query menu:

image

WP Query configuration modal:

image

WP sort modal:

image

Dates on activity tab of details pane:

image

Adding relations on old wp show page:

image

Adding watchers on old wp show page:

image

Timeline edit page:

image

Calendar filtering:

image

Repository revision jump box:

image

Repository revision diff view selector:

image

Wiki create new child page form:

image

New status reporting form:

image

New dependency form:

image

Project settings module selection:

image

Admin project filter:

image

Admin user filtering:

Same as above (buttons not aligned)

Admin workflows role/type selection:

image

Admin custom field edit:

image

Admin colors form:

image

My page edit block selection (has become unusable because of the overlapping):

image

Login screen:

image

Cost reporting plugin filter:

image

As a general note: we should probably limit the max width of input fields as they otherwise tend to look strange.

**Comment by myabc** [comment]
@ulferts 

RE:

> Timeline toolbar:
> Calendar filtering:
> Admin project filter:
> Admin user filtering:

Do you have the latest commits?

RE:

> Filters on activity page:
> Admin group user selection:
> Admin custom field edit:

Please make sure you're on Foundation for Apps 1.0.2. This new release [improves checkbox styling](https://github.com/zurb/foundation-apps/blob/master/CHANGELOG.md#version-102).

**Comment by myabc** [comment]
@ulferts 

> WP sort modal:

Only seeing this at certain breakpoints. Would prefer to migrate to ui-select before fixing this.

**Comment by ulferts** [comment]
@myabc sorry again for not having used the latest commit. I'll edit my initial comment and remove everything that no longer applies.

**Comment by myabc** [comment]
@ulferts issues in plugins should now have been addressed.

**Comment by ulferts** [comment]
Sorry for spotting this so late but the meetings plugin also has one problem:

image

**Comment by myabc** [comment]
@ulferts meetings is improved – time/date is still on lines though. This is not something I can fix without further refactoring of our forms CSS.

**Comment by myabc** [comment]
@ulferts I dealt with _most_ but _not all_ of the issues you mentioned. I'd like to break the rest out into separate WP/PRs.

**Comment by myabc** [comment]
@ulferts I've created a separate work package for form refactoring (this will also resolve the issue with upload fields).
